lib/suggest.js

```diff
@@ -302,7 +304,23 @@ function generate(context, memory, words, edits) {
       nextNextAfter = nextAfter.slice(1)
       character = nextCharacter
       nextCharacter = word.charAt(position + 1)
-      upper = character.toLowerCase() !== character
+      upper = nextUpper
+      if (nextCharacter) {
+        nextUpper = nextCharacter.toLowerCase() !== nextCharacter
+      }
+
+      if (nextAfter && upper !== nextUpper) {
+        // Remove.
+        check(before + switchCase(nextAfter))
+
+        // Switch.
+        check(
+          before +
+            switchCase(nextCharacter) +
+            switchCase(character) +
+            nextNextAfter
+        )
```

Comment:
These remove and switch in case-specific ways when the case of the current character differs from the case of the next character.

Without this extra logic, the "Ghandi" test would fail by missing "Hanoi", "Hindi" and "Randi" and by returning "hand", "hands", and "handy" in lowercase.